react-carousel-component/my-app/src/Carousel.js

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from 'react';
+import PrevButton from './PrevButton';
+import NextButton from './NextButton';
+import Content from './Content';
+import Indicators from './Indicators';
+import './Carousel.css';
+
+export default function Carousel({ images }) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  useEffect(() => {
+    const interval = setInterval(handleClickNext, 3000);
+    return () => clearInterval(interval);
+  });
```

Comment:
Where is your dependency array? Without it, this effect will run on _every_ render, which means that on every render the clean up function will get called and the interval will be cleared, then you'll start it again. So the 3 second interval will start over on each render. That is not what you want!

Reply:
In line 11, useEffect includes function defined outside its scope, so I added handleClickNext in the dependency array.